# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: widget/CompositorWidget.h
# Commit: 7ac989d08efb
# Full Hash: 7ac989d08efb12bad409cc29984c58c56a04a6b1
# Author: stransky <stransky@redhat.com>
# Date: 2021-01-22 17:29:50
# Regressor Bug: 1674473
# File Overlap Count: 1
# Description:
#   Bug 1674473 [Linux] SW-WR Clear background when painting popup windows, r=lsalzman
#   
#   Clear alpha from buffer before painting transparent windows by SW Webrender to avoid rendering artifacts.
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D102189
# ==============================================================================

diff -r 3046f61ecff2 -r 7ac989d08efb widget/CompositorWidget.h
--- a/widget/CompositorWidget.h	Fri Jan 22 08:10:12 2021 +0000
+++ b/widget/CompositorWidget.h	Wed Jan 20 21:11:48 2021 +0000
@@ -164,6 +164,13 @@
   virtual bool NeedsToDeferEndRemoteDrawing() { return false; }
 
   /**
+   * Some widgets (namely Gtk) may need clean up underlying surface
+   * before painting to draw transparent objects correctly.
+   */
+  virtual void ClearBeforePaint(RefPtr<gfx::DrawTarget> aTarget,
+                                const LayoutDeviceIntRegion& aRegion) {}
+
+  /**
    * Called when shutting down the LayerManager to clean-up any cached
    * resources.
    *